### PR TITLE
Fix 503s part 1: add validate_cluster=true 

### PR DESCRIPTION
**Pull request by andraxylia**
Opened a new PR instead of #3482 

Added validate_cluster option that can be controlled with an env var called PILOT_VALIDATE_CLUSTERS.

### CODE REVIEW

File: pilot/pkg/proxy/envoy/v1/config.go [hunk 1]
```diff
@@ -35,6 +35,12 @@ import (
 	"istio.io/istio/pkg/log"
 )
 
+var (
+	// validateClustersDefault is an environment variable that can be set to false to disable
+	// cluster validation in RDS, in case problems are discovered.
+	ValidateClustersDefault = true
```
Comment (ldemailly) [inline pilot/pkg/proxy/envoy/v1/config.go]: why is it called “default” when it’s going to be the value used, is there a way to set a different value that what get sets into this var (by the env)

File: pilot/pkg/proxy/envoy/v1/config.go [hunk 2]
```diff
@@ -35,6 +35,12 @@ import (
 	"istio.io/istio/pkg/log"
 )
 
+var (
+	// validateClustersDefault is an environment variable that can be set to false to disable
```
Comment (ldemailly) [inline pilot/pkg/proxy/envoy/v1/config.go]: Comment should start uppercase (how come the linters don't complain?)

I propose you name it "ValidateClusters" drop the misleading "Default"

Comment (ZackButcher) [reply pilot/pkg/proxy/envoy/v1/config.go]: Relevant section of [Effective Go](https://golang.org/doc/effective_go.html#commentary):

> If every doc comment begins with the name of the item it describes, the output of godoc can usefully be run through grep. Imagine you couldn't remember the name "Compile" but were looking for the parsing function for regular expressions, so you ran the command,

Comment (andraxylia) [reply pilot/pkg/proxy/envoy/v1/config.go]: Yup, linter got it.

**Comment by ldemailly** [comment]
Thanks for the name clarification/change!

I don't think this should go in without : 

> let’s add a test (feel free to copy/adapt #1041 )

but feel free to do it anyway (remember this is fixit week)

ps: pr still doesn't pass linter

**Pull request merged**